code/game/objects/items/tools/weldingtool.dm

```diff
@@ -122,12 +127,15 @@
 
 	if(affecting && (!IS_ORGANIC_LIMB(affecting)) && user.a_intent != INTENT_HARM)
 		if(src.use_tool(H, user, 0, volume=50, amount=1))
+			H.add_overlay(GLOB.welding_sparks)
 			if(user == H)
 				user.visible_message("<span class='notice'>[user] starts to fix some of the dents on [H]'s [parse_zone(affecting.body_zone)].</span>",
 					"<span class='notice'>You start fixing some of the dents on [H == user ? "your" : "[H]'s"] [parse_zone(affecting.body_zone)].</span>")
 				if(!do_mob(user, H, 50))
+					H.cut_overlay(GLOB.welding_sparks)
 					return
 			item_heal_robotic(H, user, 15, 0)
+			H.cut_overlay(GLOB.welding_sparks)
```

Comment:
what youre doing here seems confusing. why not just use use_tool for the fixing chassis?
you can replace the use_tool being used for fuel consumption with use() and tool_start_check()

Reply:
Are you suggesting for me to rewrite the code that was previously written because I'm implementing a new feature in it?

Reply:
yeah you are editing a proc in a way that can be done better with a tiny rewrite so im asking you to do so

Reply:
ah, sure, was just sitting for a while trying to figure out how am I supposed to do that without doing that, nvm lol

Reply:
Changed the code a bit.
I use use_tool because it does the delay, sound and vfx. I don't see a reason to use use instead of use_tool, as I would have to implement a few things that use_tool does in order to have the same functionality of that tool on usage.